openedx/openedx-platform
# feat: Copy tags when sync library [FC-0062]

## Description

This PR adds:

- Call `copy_tags` function of tagging API to copy tags from all library components when adding (copy) a library in a course.
- Update `ObjectTagMinimalSerializer` to avoid delete tags that are copied.
- Which edX user roles will this change impact? "Course Author"

## Supporting information

- Github issue: https://github.com/openedx/modular-learning/issues/177
- Depends on: https://github.com/openedx/openedx-learning/pull/236
- Internal ticket: [FAL-3616](https://tasks.opencraft.com/browse/FAL-3616)

## Testing instructions

- Run this script to create taxonomies: https://github.com/open-craft/taxonomy-sample-data
- Run `make requirements` in lms and cms shell or rebuild your openedx-dev image in tutor.
- To test this you need to use the legacy Unit page (disable [contentstore.new_studio_mfe.use_new_unit_page](http://studio.local.edly.io:8001/admin/waffle/flag/25/change/))
- Create a new library with components. Add tags to these components.
- Copy to Clipboard a component
- Go to a unit inside a course.
- Paste the component. Refresh the page to see tags count button.
- Open the manage tag, verify the copied tags. Edit and verify that you can't delete.
- Add more tags and verify that you can delete the new added tags.
- On the library, add more tags to the component. Edit the component and update the title.
- Return to the course and refresh the page.
- Click on "Update Available" of the new component.
- Verify the new tags.
- On the library add the same tags of the component in the course. Edit the component and update the title.
- Return to the course and refresh the page. Click on "Update Available" of the new component.
- Verify the new tags and verify that you cant delete the tags

## Deadline

ASAP

## Other information

Before merge:

- [x] Bump `openedx-learning` version: https://github.com/openedx/openedx-learning/pull/236
